### ISSUE TITLE
Out of sync with latest released version

### ISSUE THREAD

**Issue by fulv** [devrafalko/webpack-karma-jasmine#1]
There are a slew of problems that would punish an unwitting user simply for trying to follow the instructions in the readme.

To start, `/sample/package.json` has an outdated version pin: 

        "webpack-karma-jasmine": "^2.0.1"

Whereas the latest version on npm is 3.0.0.

Next, when you run `npm install` in `webpack-karma-jasmine/sample`, you get;

    npm WARN karma-webpack@3.0.5 requires a peer of webpack@^2.0.0 || ^3.0.0 but none is installed. You must install peer dependencies yourself.

and 

    found 6 vulnerabilities (1 low, 5 moderate)
    run `npm audit fix` to fix them, or `npm audit` for details

Finally, `npm test` doesn't work at all:

```
> npm test

> karma-sample@1.0.0 test /home/me/javascript/webpack-karma-jasmine/sample
> karma start

(node:9315) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
ℹ ｢wdm｣: wait until bundle finished: noop
14 09 2018 21:26:28.072:ERROR [karma]: TypeError: Path must be a string. Received undefined
    at assertPath (path.js:28:11)
    at Object.join (path.js:1236:7)
    at Plugin.<anonymous> (/home/me/javascript/webpack-karma-jasmine/sample/node_modules/karma-webpack/lib/karma-webpack.js:274:66)
    at Plugin.readFile (/home/me/javascript/webpack-karma-jasmine/sample/node_modules/karma-webpack/lib/karma-webpack.js:294:5)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
✖ ｢wdm｣: 
ERROR in chunk tests/spec_a [entry]
bundled.js
Conflict: Multiple chunks emit assets to the same filename bundled.js (chunks 0 and 1)
ℹ ｢wdm｣: Failed to compile.
npm ERR! Test failed.  See above for more details.
```

Sadly, updating the version pin in `/sample/package.json` to `3.0.0` does not really improve things, except that at least then the vulnerabilities are gone.

**Comment by devrafalko** [comment]
Thank you for your remarks. I've considered all of them in the new version. It should work fine now.

**Comment by fulv** [comment]
Wow! You rock!  This is awesome.  Not just because you fixed everything, but also, not being very experienced with this stack myself, because I have learned a lot from your commits.  I'm very grateful.
TBH, I felt a bit bad for perhaps not phrasing my issue very constructively, so I was not expecting such a quick resolution.  Thanks!
Feel free to close this issue.

**Comment by fulv** [comment]
I've created a pull request to fix a few small typos in the readme and the karma.conf.js.  Also, to upgrade the version in sample/package.json.  

Would it be ok to upgrade `"webpack": "^4.19.0"`?

Finally, even though it's working now, there is still a warning when running `npm i`:

    npm WARN karma-webpack@3.0.5 requires a peer of webpack@^2.0.0 || ^3.0.0 but none is installed. You must install peer dependencies yourself.

What would be a good way to fix this?

**Comment by devrafalko** [comment]
The `^` character in the dependency version means that the latest **minor** version will be installed so `"webpack": "^4.16.0"` will install `"4.19.0"` version anyway.

The `karma-webpack@3.0.5 requires a peer of webpack@^2.0.0 [...]` warning is an issue webpack-contrib/karma-webpack#358 here. I've upgraded the `karma-webpack` to the `next` version.

I've already merged your pull request. Thank you for your commitment!
